site/src/modules/apps/useAppLink.ts

```diff
@@ -0,0 +1,75 @@
+import { apiKey } from "api/queries/users";
+import type {
+	Workspace,
+	WorkspaceAgent,
+	WorkspaceApp,
+} from "api/typesGenerated";
+import { displayError } from "components/GlobalSnackbar/utils";
+import { useProxy } from "contexts/ProxyContext";
+import type React from "react";
+import { useQuery } from "react-query";
+import {
+	getAppHref,
+	isExternalApp,
+	needsSessionToken,
+	openAppInNewWindow,
+} from "./apps";
+
+type UseAppLinkParams = {
+	workspace: Workspace;
+	agent: WorkspaceAgent;
+};
+
+export const useAppLink = (
+	app: WorkspaceApp,
+	{ agent, workspace }: UseAppLinkParams,
+) => {
+	const label = app.display_name ?? app.slug;
+	const { proxy } = useProxy();
+	const { data: apiKeyResponse } = useQuery({
+		...apiKey(),
+		enabled: isExternalApp(app) && needsSessionToken(app),
```

Comment:
Could be `enabled: needsSessionToken(app),` since `needsSessionToken` already checks `isExternalApp`.